homeassistant/components/sensor/google_travel_time.py

```diff
@@ -23,47 +25,97 @@
 CONF_ORIGIN = 'origin'
 CONF_DESTINATION = 'destination'
 CONF_TRAVEL_MODE = 'travel_mode'
+CONF_OPTIONS = 'options'
+CONF_MODE = 'mode'
+CONF_NAME = 'name'
 
 PLATFORM_SCHEMA = vol.Schema({
     vol.Required('platform'): 'google_travel_time',
+    vol.Optional(CONF_NAME): vol.Coerce(str),
     vol.Required(CONF_API_KEY): vol.Coerce(str),
     vol.Required(CONF_ORIGIN): vol.Coerce(str),
     vol.Required(CONF_DESTINATION): vol.Coerce(str),
     vol.Optional(CONF_TRAVEL_MODE, default='driving'):
-        vol.In(["driving", "walking", "bicycling", "transit"])
+        vol.In(["driving", "walking", "bicycling", "transit"]),
+    vol.Optional(CONF_OPTIONS): vol.All(
+        dict, vol.Schema({
+            'mode': cv.string,
+            'language': cv.string,
+            'avoid': cv.string,
+            'units': cv.string,
+            'arrival_time': cv.string,
+            'departure_time': cv.string,
+            'traffic_model': cv.string,
+            'transit_mode': cv.string,
+            'transit_routing_preference': cv.string
+        }))
 })
 
 
+def convert_time_to_utc(timestr):
+    """Take a string like 08:00:00 and convert it to a unix timestamp."""
+    return dt_util.as_timestamp(datetime.combine(dt_util.start_of_local_day(),
+                                                 dt_util.parse_time(timestr)))
+
+
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup the travel time platform."""
     # pylint: disable=too-many-locals
+    options = config.get(CONF_OPTIONS)
+    departure_time = options.get('departure_time')
+    arrival_time = options.get('arrival_time')
+    if departure_time is not None and departure_time is not 'now':
+        options['departure_time'] = convert_time_to_utc(departure_time)
+    else:
+        options['departure_time'] = 'now'
+
+    if arrival_time is not None and arrival_time is not 'now':
+        options['arrival_time'] = convert_time_to_utc(arrival_time)
+    else:
+        options['arrival_time'] = 'now'
+
+    departure_time = options.get('departure_time')
+    arrival_time = options.get('arrival_time')
+    if departure_time is not None and arrival_time is not None:
+        del options['arrival_time']
```

Comment:
Maybe print a warning to tell the user that both arrival and departure time can not be specified?